### PR TITLE
test + fix: regression corpus for real-world parse failures; fix 4 low-tier gaps

### PR DESCRIPTION
**Pull request by tinovyatkin**
Adds a regression-test corpus file (`test/corpus/regressions.txt`) containing
18 real-world PowerShell patterns that currently fail to parse, then fixes
the four low-tier gaps that could be addressed with narrow grammar tweaks.
The remaining 14 gaps are tracked in dedicated issues (#27–#38) and stay
`:skip`-ed in the regression corpus with comments pointing to each issue.

## Fixed

| # | Test name | Real-world source |
|---|---|---|
| ✅ | Empty scriptblock literal as right-hand side of assignment | gregoryfmartin/Eldoria/.../MapTileObject.ps1 |
| ✅ | Assembly-qualified type literal | PowerShell/PowerShell test/powershell/.../Parsing.Tests.ps1 |
| ✅ | Function name with symbolic leading char | nightroman/Invoke-Build/Invoke-Build.ps1 |
| ✅ | Unix absolute-path command invocation | adamdriscoll/selenium-powershell/Internal/init.ps1 |

Each fix ships as its own commit with a minimal grammar change, a
regenerated `parser.c` / `grammar.json` / `node-types.json` /
`parser.h` / `array.h` / `node_types.go`, and the corresponding
regression test unskipped with its captured expected tree.

### Grammar changes

- `script_block_expression`: make the inner `script_block` optional so
  `{}` parses as an expression.
- `type_literal`: accept an optional `assembly_qualifier` tail
  (`, AssemblyName`). New nodes: `assembly_qualifier`, `assembly_name`.
- `command_name`: drop `*` from the leading-char exclusion set so
  function calls like `*Die 'x' 1` parse.
- `command_name` / `path_command_name_token`: allow `/` so Unix-style
  absolute-path invocations parse (`/usr/bin/stat -f "%A" "/tmp"`).

## Deferred to dedicated issues

Three items originally estimated as "low-tier easy" turned out to hit
the same structural issues tracked in the higher-tier bugs:

- **Function body with only an attribute then a comment** —
  requires GLR disambiguation between `attribute_list` contexts
  (`param` / `class` / `enum` / `script_parameter`) and a bare
  attribute-as-expression. See #33 (related class).
- **Unquoted `[!]` as command argument** — structurally identical to
  the element-access / type-literal vs bareword ambiguity. Tracked
  in #33.
- **Adjacent `(expr).Member""` as a single argument** — adjacent-token
  concatenation case, part of the token-tokenizer work tracked in #27.

Each deferred test has an inline comment in `regressions.txt`
explaining why it remains `:skip`-ed.

## Validation

- `npx tree-sitter test`: **227 / 227 passing** (223 existing + 4 newly
  unskipped regressions).
- `test/fixtures/*`: every script still parses clean (no ERROR /
  MISSING nodes).
- `npm run generate`: ABI 14, no conflicts beyond pre-existing ones.

## Related

The top-tier and mid-tier gaps discovered during the same survey are
tracked separately: #27, #28, #29, #30, #31, #32, #33, #34, #35, #36,
#37, #38.

## Summary by CodeRabbit

* **New Features**
  * Support for assembly-qualified type syntax inside brackets (e.g., `[Type, Assembly]`)
  * Broader command/path name parsing to accept additional characters (improved handling of symbols and path separators)
  * Script block expressions may now be empty

* **Tests**
  * Added a comprehensive regression suite covering type literals, native/-path commands, script blocks, and other parsing edge cases

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1431,6 +1432,13 @@ export default grammar({
         $.type_name,
       ),
 
+    // Assembly-qualified type name tail at the outermost `[...]`, e.g.
+    // `[Int32, mscorlib]` or
+    // `[System.Collections.Generic.List[string], System.Private.CoreLib]`.
+    assembly_qualifier: ($) => seq(',', $.assembly_name),
+
+    assembly_name: ($) => $.type_name,
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: Addressed in 505e1e1: `assembly_name` now accepts `type_name` followed by zero-or-more `, Key=Value` metadata entries (new `assembly_attribute` / `assembly_attribute_name` / `assembly_attribute_value` nodes), so forms like

```powershell
[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]
```

parse. Values are constrained to the identifier / dotted-version / hex-token shapes that `AssemblyName.FullName` actually emits; quoted values are intentionally excluded because real PowerShell rejects `Custom="quoted"` too (verified via `[System.Management.Automation.Language.Parser]::ParseInput`).

The regression test `Assembly-qualified type literal` in `test/corpus/regressions.txt` now covers the full-AQN case.

File: grammar.js [hunk 2]
```diff
@@ -1219,7 +1219,7 @@ export default grammar({
       seq('@(', field('statements', optional($.statement_list)), ')'),
 
     script_block_expression: ($) =>
-      seq('{', optional($.param_block), $.script_block, '}'),
+      seq('{', optional($.param_block), optional($.script_block), '}'),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js:1222]: Tested the suggestion — it regresses the existing corpus test `unicode_identifiers` (`Unicode identifiers and decimal digits`) on the case:

```powershell
$π = { param($θ) $θ * 3.14 }
```

which is valid PowerShell (confirmed via `[System.Management.Automation.Language.Parser]::ParseInput`). The reason is that `script_block`'s param-block branch requires a `_statement_boundary` (explicit `;` or newline) between `param(...)` and the body, whereas the form `{ param($x) body }` with just whitespace is legal. Keeping `optional($.param_block)` inside `script_block_expression` lets this form parse as `(param_block)(script_block with body only)`.

On the ambiguity concern: `tree-sitter generate` emits no LALR/GLR conflict warnings for this rule, and the parser consistently picks the `(param_block)(script_block-body-only)` shape — not two competing shapes — because the second branch of `script_block` requires the boundary and therefore does not match in that context.

The cleaner restructuring would be to make `_statement_boundary` optional in `script_block`'s param-block branch (or move param-block handling entirely into `script_block_expression`), but that is a broader refactor with its own ripple effects; deferring for now.
